test/e2e/framework/util.go

```diff
@@ -2406,11 +2406,19 @@ func WaitForStableCluster(c clientset.Interface, masterNodes sets.String) int {
 	}
 	allPods.Items = currentPods
 	scheduledPods, currentlyNotScheduledPods := e2epod.GetPodsScheduled(masterNodes, allPods)
-	for len(currentlyNotScheduledPods) != 0 {
+	systemPods, err := c.CoreV1().Pods(metav1.NamespaceSystem).List(metav1.ListOptions{})
+	ExpectNoError(err, "listing all pods in kube-system namespace while waiting for stable cluster")
+	_, currentlyNotScheduledSystemPods := e2epod.GetPodsScheduled(masterNodes, systemPods)
+
+	// Wait for system pods to be scheduled, and for pods in all other namespaces to be deleted
+	for len(currentlyNotScheduledPods) != 0 && (len(currentlyNotScheduledSystemPods) != 0 || len(systemPods.Items) != len(allPods.Items)) {
```

Comment:
why do we need ```len(currentlyNotScheduledPods) != 0 &&```? I think the remaining part should be sufficient: it tests that all system pods are scheduled, and that all scheduled pods are system pods.

Reply:
True, updated